Kernel/FileSystem/ISO9660FileSystem.h

```diff
@@ -0,0 +1,395 @@
+    VolumeDescriptorHeader header;
+    u8 boot_system_identifier[32];
+    u8 boot_identifier[32];
+    u8 boot_system_use[1977];
+};
+static_assert(sizeof(BootRecord) == 2048);
+
+// 8.3 Volume Descriptor Set Terminator
+struct [[gnu::packed]] VolumeDescriptorSetTerminator {
+    VolumeDescriptorHeader header;
+    u8 zeros[2041];
+};
+static_assert(sizeof(VolumeDescriptorSetTerminator) == 2048);
+
+// 8.4 Primary Volume Descriptor
+struct [[gnu::packed]] PrimaryVolumeDescriptor {
+    VolumeDescriptorHeader header;
+    u8 unused1;
+    u8 system_identifier[32];
+    u8 volume_identifier[32];
+    u64 unused2;
+    LittleAndBigEndian<u32> volume_space_size;
+    u8 unused3[32];
+    LittleAndBigEndian<u16> volume_set_size;
+    LittleAndBigEndian<u16> volume_sequence_number;
+    LittleAndBigEndian<u16> logical_block_size;
+    LittleAndBigEndian<u32> path_table_size;
+
+    u32 l_path_table_occurence_location;
+    u32 l_path_table_optional_occurence_location;
+    u32 m_path_table_occurence_location;
+    u32 m_path_table_optional_occurence_location;
+
+    DirectoryRecordHeader root_directory_record_header;
+    u8 root_directory_identifier; // Exactly 0x00.
+
+    u8 volume_set_identifier[128];
+    u8 publisher_identifier[128];
+    u8 data_preparer_identifier[128];
+    u8 application_identifier[128];
+
+    u8 copyright_file_identifier[37];
+    u8 abstract_file_identifier[37];
+    u8 bibliographic_file_identifier[37];
+
+    AsciiDateAndTime volume_creation_date_and_time;
+    AsciiDateAndTime volume_modification_date_and_time;
+    AsciiDateAndTime volume_expiration_date_and_time;
+    AsciiDateAndTime volume_effective_date_and_time;
+
+    u8 file_structure_version; // Always 0x01.
+    u8 unused4;
+    u8 application_use[512];
+    u8 reserved[653];
+};
+static_assert(sizeof(PrimaryVolumeDescriptor) == 2048);
+
+// 8.6 Volume Partition Descriptor
+struct [[gnu::packed]] VolumePartitionDescriptor {
+    VolumeDescriptorHeader header;
+    u8 unused;
+
+    u8 system_identifier[32];
+    u8 volume_partition_identifier[32];
+    LittleAndBigEndian<u32> volume_partition_location;
+    LittleAndBigEndian<u32> volume_partition_size;
+
+    u8 system_use[1960];
+};
+static_assert(sizeof(VolumePartitionDescriptor) == 2048);
+
+}
+
+enum class RecursionDecision {
+    Recurse,
+    Continue,
+    Break
+};
+
+class ISO9660Inode;
+class ISO9660DirectoryIterator;
+
+class ISO9660FS final : public BlockBasedFileSystem {
+    friend ISO9660Inode;
+    friend ISO9660DirectoryIterator;
+
+public:
+    struct DirectoryEntry : public RefCounted<DirectoryEntry> {
+        u32 extent { 0 };
+        u32 length { 0 };
+
+        // NOTE: This can never be empty if we read the directory successfully.
+        //       We need it as an OwnPtr to default-construct this struct.
+        OwnPtr<KBuffer> blocks;
+
+        static KResultOr<NonnullRefPtr<DirectoryEntry>> create(u32 extent, u32 length, OwnPtr<KBuffer> blocks)
```

Comment:
Please call functions that attempt to create something with the possibility of failure `try_create`